docs-new/README.md

```diff
@@ -0,0 +1,123 @@
+
+* Python >=3.9
+* pip
+
+
+## Build process
+
+The directory structure is intended to mimic the tree hierarchy of the website. This will enable contributors to find the documentation they need to update easier. The static and documentation will reside in the same location. 
+
+All static pages are all the `./home/.*md` files and the docs are the `.home/docs/<version>/*.md` files. Notice the location of the `mkdocs.yml`.
+
+Looking at this though, you may ask where the older versions and javadocs are.
+
+```
+.
+├── home
+│   ├── assets
+│   │   └── images
+│   ├── docs
+│   │   └── latest
+│   │       ├── assets
+│   │       ├── api.md
+│   │       ├── aws.md
+│   │       ├── branching.md
+│   │       ├── ...
+│   │       ├── **mkdocs.yml(docs)** 
+│   │       ├── ...
+│   │       ├── spark-structured-streaming.md
+│   │       ├── spark-writes.md
+│   │       └── table-migration.md
+│   ├── ASF.md
+│   ├── about.md
+│   ├── ...
+│   ├── vendors.md
+│   └── view-spec.md
+├── LICENSE
+├── README.md
+├── **mkdocs.yml(static)**
+├── requirements.txt
+└── variables.yml
+```
+
+All of the documentation versions are saved in special `docs-<version>` tags that only contain the root of the docs version they contian. There is also a `javadoc` tag that contains all prior versions of the javadocs in a single tag. These are generated and loaded only at build time using the [git-worktree](https://git-scm.com/docs/git-worktree) docs.
+
+```
+.
+├── home
+│   ├── docs
+│   │   ├── latest
+│   │   ├── 1.3.1
+│   │   ├── 1.3.0
+│   │   └── ...
+│   ├── javadoc
+│   │   ├── latest
+│   │   ├── 1.3.1
+│   │   ├── 1.3.0
+│   │   └── ...
+```
+
+## Install
+
+```
+python -m venv mkdocs_env
+source mkdocs_env/bin/activate
+
+pip install -r requirements.txt
+```
+
+## Add git worktrees
+
+For now, I'm just adding a single version
+
+```
+git worktree add home/docs/1.3.1 docs-1.3.1
+git worktree add home/javadoc javadoc
+```
+
+## Build
+
+```
+mkdocs build
+mkdocs serve
+```
+
+## Validate Links
+
+Due to the delayed aggregation of subdocs of `mkdocs-monorepo-plugin` there are some warnings that display for subdocs that mkdocs are not able to connect due to being off by a single directory when the version directory is added, which doesn't mirror the directory layout on disk.
+
+To ensure the links work, I am temporarily using a tool called linkchedker, which will traverse the links on the livesite. 
+```
+pip install linkchecker
+
+./linkchecker http://localhost:8000 -r1 -Fcsv/link_warnings.csv
+```
+
+## Things to consider
+
+ - Do not use static links to the public iceberg site.
+ - Only use relative links. If you want to reference the root (the directory where the main mkdocs.yml is located `home` in our case) use "spec.md" vs "/spec.md". Also, static sites should only reference the `docs/*` (see next point), but docs can reference the static content normally (e.g. `branching.md` page which is a versioned page linking to `spec.md` which is a static page).
```

Comment:
Can the static pages link to docs in `docs/latest`?

Reply:
Yeah, static pages can link down to docs fine, it's linking back up that requires "../../spec.md" vs "spec.md".

Unfortunately the way you link is dependent on the context of which doc you are writing in. I tried to make this consistent across all areas of the project, but couldn't find a solution that worked with monorepo.